### ISSUE TITLE
Embedding a resource with an array attribute is not faithful

### ISSUE THREAD

**Issue by oco-adam** [ash-project/ash_postgres#126]
**Describe the bug**
A clear and concise description of what the bug is. If you are not sure if the bug is related to `ash` or an extension, log it with [ash](https://github.com/ash-project/ash/issues/new) and we will move it.

Suppose I have an embedded resource with a nullable array attribute. Then if this resource is embedded in a parent resource, then when it is persisted in the database with this value set to `nil`, then when it is extracted it changes to an empty array `[]`.

**To Reproduce**

```elixir
  defmodule EmbeddedResource do
    use Ash.Resource,
      data_layer: :embedded

    attributes do
      attribute :list_att, {:array, :string} do
        allow_nil? true
        default nil
      end
    end
  end

  defmodule MyResource do
      use Ash.Resource,
    data_layer: AshPostgres.DataLayer,
    extensions: []

  postgres do
    repo(MyRepo)
    table "my_resources"
  end

    actions do
      defaults [:read, :destroy, :create, :update]
    end

    attributes do
      uuid_primary_key :id

      attribute :embedded_resource, EmbeddedResource
    end
  end
  ```

then I would expect the following to hold:

```elixir
test "embeds and extracts faithfully" do
    embedded = %EmbeddedResource{list_att: nil}
    data = %{embedded_resource: embedded}

    %MyResource{embedded_resource: extracted} =
      MyResource
      |> Ash.Changeset.for_create(:create, data)
      |> MyApi.create!()

    assert embedded.list_att == extracted.list_att
  end
```

**Expected behavior**

I expect the value to be `nil` once extracting from the db.

** Runtime
 - Elixir version
 0.14
 OTP 25
 - OS
 macOS
 - Ash version
 2.4.26
 ash_postgres 1.2

**Comment by zachdaniel** [comment]
Hey there! Thanks for the detailed report! I've tried to reproduce this in `main` with this test here: https://github.com/ash-project/ash_postgres/blob/main/test/embeddable_resource_test.exs#L22 which appears to be doing exactly what you're doing, but that test passes for me. Can you take a look at the test and see if you can spot a difference? Or perhaps try out using ash main and see if it works for you?

**Comment by oco-adam** [comment]
Hi @zachdaniel - my apologies, I've just checked this again and it seems fine.

I am actually still seeing this issue, but now I think this might be because I'm using Ash.Generator in my real-life use case, and I simplified this to remove Ash.Generator in my minimal example above. 

So I'll close this issue because it seems like it was a red herring, either for an issue with Ash.Generator in disguise, or more likely, an issue with my understanding of how to use Ash.Generator.

So I'll look into these other issues on my end, but for now, thanks very much for looking into this and for the resolution!
